internal/resolution/variablesources/installed_package.go

```diff
@@ -73,7 +74,19 @@ func MakeInstalledPackageVariables(
 		})
 		installedBundle := resultSet[0]
 
-		upgradeEdges, err := successors(allBundles, installedBundle)
+		channels := filter.Filter[catalogmetadata.Channel](allChannels, filter.And[catalogmetadata.Channel](
+			func(entity *catalogmetadata.Channel) bool {
+				return entity.Package == clusterExtension.Spec.PackageName
+			},
+			func(entity *catalogmetadata.Channel) bool {
+				if clusterExtension.Spec.Channel != "" {
+					return entity.Name == clusterExtension.Spec.Channel
+				}
+				return true
+			},
+		))
```

Comment:
Wondering if we can build a map of package -> channel once and re-used it multiple times to avoid iterating trough all the channel date for each call to this function.

Reply:
We certainly could, but I wasn't sure if that optimization is worth the overhead / additional complexity within the client logic. I'm definitely open to discussing optimization in this PR, but also want to make sure we aren't prematurely trying to optimize and dig ourselves into a different hole of complexity. Especially since there has been mention of potentially not even needing to have a sat solver that could result in needing to refactor a large chunk of this logic again.